File: mobile/lib/models/category.dart
class Category {
  final String id;
  final String name;
  final String? iconName;
  final int sortOrder;

  Category({
    required this.id,
    required this.name,
    this.iconName,
    required this.sortOrder,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'] as String,
      name: json['name'] as String,
      iconName: json['iconName'] as String?,
      sortOrder: json['sortOrder'] as int,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'iconName': iconName,
      'sortOrder': sortOrder,
    };
  }
}

File: backend/src/Pawora.Core/Entities/Category.cs
namespace Pawora.Core.Entities;

public class Category : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? IconName { get; set; }
    public int SortOrder { get; set; }

    public ICollection<Product> Products { get; set; } = new List<Product>();
}
